static/scripts/rendering/config-editor.ts

```diff
@@ -0,0 +1,109 @@
+import { Manifest, Plugin } from "../../types/plugins";
+import { controlButtons } from "./control-buttons";
+import { ManifestRenderer } from "../render-manifest";
+import { processProperties } from "./input-parsing";
+import { updateGuiTitle } from "./utils";
+import { writeNewConfig } from "./write-add-remove";
+import MarkdownIt from "markdown-it";
+import { getManifestCache } from "../../utils/storage";
+const md = new MarkdownIt();
+
+export function renderConfigEditor(renderer: ManifestRenderer, pluginManifest: Manifest | null, plugin?: Plugin["uses"][0]["with"]): void {
+  renderer.currentStep = "configEditor";
+  renderer.backButton.style.display = "block";
+  renderer.manifestGuiBody.innerHTML = null;
+  controlButtons({ hide: false });
+  processProperties(renderer, pluginManifest?.configuration?.properties || {});
+  const configInputs = document.querySelectorAll<HTMLInputElement | HTMLTextAreaElement>(".config-input");
+
+  if (plugin) {
+    configInputs.forEach((input) => {
+      const key = input.getAttribute("data-config-key");
+      if (!key) {
+        throw new Error("Input key is required");
+      }
+
+      const keys = key.split(".");
+      let currentObj = plugin;
+      for (let i = 0; i < keys.length; i++) {
+        if (!currentObj[keys[i]]) {
+          break;
+        }
+        currentObj = currentObj[keys[i]] as Record<string, unknown>;
+      }
+
+      let value: string;
+
+      if (typeof currentObj === "object") {
+        value = JSON.stringify(currentObj, null, 2);
+      } else {
+        value = currentObj as string;
+      }
+
+      if (input.tagName === "TEXTAREA") {
+        (input as HTMLTextAreaElement).value = value;
+      } else {
+        (input as HTMLInputElement).value = value;
+      }
+    });
+  }
+
+  const add = document.getElementById("add") as HTMLButtonElement;
+  const remove = document.getElementById("remove") as HTMLButtonElement;
+  if (!add || !remove) {
+    throw new Error("Add or remove button not found");
+  }
+  add.addEventListener("click", writeNewConfig.bind(null, renderer, "add"));
```

Comment:
the problem is this addEventListener statement, if only you navigate a couple times and then execute an action it will stackup. you only need one event listener please check that. 

[eventListenerAdded.webm](https://github.com/user-attachments/assets/976db9d7-b09b-42c5-9663-72d10b858773)